server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java

```diff
@@ -581,9 +582,10 @@ private ReadOnlyLDAPUser searchAndBuildUser(String name) throws NamingException
       sc.setReturningAttributes(new String[] { userIdAttribute });
       sc.setCountLimit(1);
 
-      StringBuilder builderFilter = new StringBuilder("(&(");
-      builderFilter.append(userIdAttribute).append("=").append(name).append(")")
-                   .append("(objectClass=").append(userObjectClass).append(")");
+      StringBuilder builderFilter = new StringBuilder(FilterEncoder.format("(&({0}={1})(objectClass={2})",
+                                                                           userIdAttribute,
+                                                                           name,
+                                                                           userObjectClass));
```

Comment:
I guess the intent will be clearer with:
```
if (StringUtils.isNotEmpty(filter)){
  filterTemplate = "(&({0}={1})(objectClass={2})" + filter + ")"
} else {
  filterTemplate = "(&({0}={1})(objectClass={2}))"
}
String filterByUserId = FilterEncoder.format(filterTemplate, ...)

Reply:
Find better